db/write_batch.cc

```diff
@@ -53,13 +55,13 @@ Status WriteBatch::Iterate(Handler* handler) const {
   int found = 0;
   while (!input.empty()) {
     found++;
-    char tag = input[0];
+    ValueType tag = (ValueType)input[0];
     input.remove_prefix(1);
     switch (tag) {
       case kTypeValue:
         if (GetLengthPrefixedSlice(&input, &key) &&
             GetLengthPrefixedSlice(&input, &value)) {
-          handler->Put(key, value);
+            handler->Put(key, value, kTypeValue, 0);
```

Comment:
Something feels a bit "off" to call `Handler::Put()` with values (`kTypeValue` and `0`) when the method is declared to take const references. The compiler allows it, which means it is likely putting the values in temporary anonymous variables and passing references to them, but it still looks funny. Not really suggesting anything here, just observing.